src/poseidon/mod.rs

```diff
@@ -77,3 +78,27 @@ pub trait ROCircuitTrait<F: PrimeFieldBits + FromUniformBytes<64>> {
         num_bits: NonZeroUsize,
     ) -> Result<Vec<AssignedBit<F>>, Error>;
 }
+
+/// Random Oracle is represented as a pair of on-circuit & off-circuit types,
+/// allowing the use of a single generic.
+pub trait ROPair<C: CurveAffine>
```

Comment:
I don't quite understand the purpose of ROPair here. When need to ensure the results matches between on-circuit and off-circuit, we can just create one for off-circuit and one for on-circuit with same Args.

Reply:
I need to pass in IVC and {off,on}-circuit versions of oracles and I don't want to link the generics directly into IVC as that's a lot of code in the `where` section.

In fact, I'm taking some of these generics outside of IVC to a separate trait

Reply:
That is, I did the code directly first, got a very large `where` and thought how I could link these oracles more elegantly